dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/messaging/SingleDataRequestEmailMessageSender.kt

```diff
@@ -0,0 +1,97 @@
+import org.dataland.datalandmessagequeueutils.messages.TemplateEmailMessage
+import org.dataland.keycloakAdapter.auth.DatalandJwtAuthentication
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * A class that manages generating emails messages for bulk and single data requests
+ */
+@Component
+class SingleDataRequestEmailMessageSender(
+    @Autowired private val cloudEventMessageHandler: CloudEventMessageHandler,
+    @Autowired private val objectMapper: ObjectMapper,
+    @Autowired val companyApi: CompanyDataControllerApi,
+) : DataRequestEmailMessageSenderBase() {
+    /**
+     * Data structure holding the shared information of the sent messages
+     */
+    data class MessageInformation(
+        val userAuthentication: DatalandJwtAuthentication,
+        val datalandCompanyId: String,
+        val dataType: DataTypeEnum,
+        val reportingPeriods: Set<String>,
+    )
+
+    /**
+     * Function that generates the message object for single data request internal mails
+     */
+    fun sendSingleDataRequestInternalMessage(
+        messageInformation: MessageInformation,
+        correlationId: String,
+    ) {
+        val companyName = companyApi.getCompanyInfo(messageInformation.datalandCompanyId).companyName
+        val properties = mapOf(
+            "User" to messageInformation.userAuthentication.userDescription,
+            "Data Type" to messageInformation.dataType.value,
+            "Reporting Periods" to formatReportingPeriods(messageInformation.reportingPeriods),
+            "Dataland Company ID" to messageInformation.datalandCompanyId,
+            "Company Name" to companyName,
+        )
+        val message = InternalEmailMessage(
+            "Dataland Single Data Request",
+            "A single data request has been submitted",
+            "Single Data Request",
+            properties,
+        )
+        cloudEventMessageHandler.buildCEMessageAndSendToQueue(
+            objectMapper.writeValueAsString(message),
+            MessageType.SendInternalEmail,
+            correlationId,
+            ExchangeName.SendEmail,
+            RoutingKeyNames.internalEmail,
+        )
+    }
+
+    /**
+     * Function that generates the message object for single data request external mails
+     */
+    fun sendSingleDataRequestExternalMessage(
+        messageInformation: MessageInformation,
+        receiver: String,
+        contactMessage: String?,
+        correlationId: String,
+    ) {
+        val companyName = companyApi.getCompanyInfo(messageInformation.datalandCompanyId).companyName
+        val properties = mapOf(
+            "companyId" to messageInformation.datalandCompanyId,
+            "companyName" to companyName,
+            "requesterEmail" to messageInformation.userAuthentication.username,
+            "dataType" to readableFrameworkNameMapping[messageInformation.dataType]!!,
```

Comment:
Gefährlich.  Würde ich nur akzeptieren, wenn es einen unit test gibt, der sicherstellt, dass alle data types in DataTypeEnum in dieser map sind.

Ansonsten führt irgendjemand ein neues Framework ein und merget nach main, vergisst aber diese Map hier zu erweitern, und plötzlich entseht hier eine NullPointerException.